File: php/Baju.php
<?php

require('Aksesoris.php'); 

class Baju extends Aksesoris {
    protected $untuk;
    protected $size;
    protected $merk;

    public function __construct($id, $nama, $harga, $stok, $jenis, $bahan, $warna, $untuk, $size, $merk, $foto = "") {
        parent::__construct($id, $nama, $harga, $stok, $jenis, $bahan, $warna, $foto);
        $this->untuk = $untuk;
        $this->size = $size;
        $this->merk = $merk;
    }

    public function get_untuk() {
        return $this->untuk;
    }

    public function get_size() {
        return $this->size;
    }

    public function get_merk() {
        return $this->merk;
    }

    public function get_column_headers() {
        $headers = parent::get_column_headers();
        array_push($headers, "UNTUK", "SIZE", "MERK");
        return $headers;
    }

    public function get_data() {
        $data = parent::get_data();
        array_push($data, $this->untuk, $this->size, $this->merk);
        return $data;
    }
}


?>

File: cpp/Baju.cpp
#include "Aksesoris.cpp"

class Baju : public Aksesoris {
    private:
        string untuk;
        string size;
        string merk;
    
    public:
        Baju(int id, const string& nama, double harga, int stok,
             const string& jenis, const string& bahan, const string& warna,
             const string& untuk, const string& size, const string& merk)
            : Aksesoris(id, nama, harga, stok, jenis, bahan, warna) {
            this->untuk = untuk;
            this->size = size;
            this->merk = merk;
        }
    
        string getUntuk() const { return this->untuk; }
        string getSize() const { return this->size; }
        string getMerk() const { return this->merk; }
    
        vector<string> getColumnHeaders() const { 
            vector<string> headers = Aksesoris::getColumnHeaders();
            headers.push_back("UNTUK");
            headers.push_back("SIZE");
            headers.push_back("MERK");
            return headers;
        }
    
        vector<string> getData() const { 
            vector<string> data = Aksesoris::getData();
            data.push_back(this->untuk);
            data.push_back(this->size);
            data.push_back(this->merk);
            return data;
        }
    };
    